netutils/connectedhomeip/CMakeLists.txt

```diff
@@ -0,0 +1,248 @@
+# ##############################################################################
+# external/connectedhomeip/CMakeLists.txt
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more contributor
+# license agreements.  See the NOTICE file distributed with this work for
+# additional information regarding copyright ownership.  The ASF licenses this
+# file to you under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License.  You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+#
+# ##############################################################################
+
+if(CONFIG_MATTER)
+
+  include(ExternalProject)
+
+  get_filename_component(
+    CHIP_ROOT ${NUTTX_APPS_DIR}/netutils/connectedhomeip/connectedhomeip
+    REALPATH)
+  get_filename_component(GN_ROOT_TARGET ${CHIP_ROOT}/config/nuttx/chip-gn
+                         REALPATH)
+  get_filename_component(COMMON_CMAKE_SOURCE_DIR
+                         ${CHIP_ROOT}/config/common/cmake REALPATH)
+
+  include(${COMMON_CMAKE_SOURCE_DIR}/chip_gn_args.cmake)
+  include(${COMMON_CMAKE_SOURCE_DIR}/chip_gn.cmake)
+
+  set(MATTER_INCDIR
+      ${CMAKE_BINARY_DIR}/include
+      ${CMAKE_BINARY_DIR}/include_arch
+      ${CMAKE_BINARY_DIR}/apps/netutils/connectedhomeip/gen/include
+      ${NUTTX_DIR}/include/libcxx
+      ${NUTTX_DIR}/libs/libxx/libcxx/include
+      ${NUTTX_DIR}/include
+      inipp
+      nlio/include
+      nlassert/include
+      connectedhomeip/src
+      connectedhomeip/src/include
+      connectedhomeip/src/lib/dnssd
+      connectedhomeip/src/app/util/mock/include
+      connectedhomeip/zzz_generated
+      connectedhomeip/zzz_generated/lighting-app
+      connectedhomeip/zzz_generated/app-common
+      connectedhomeip/src/platform/Linux
+      connectedhomeip/examples/platform/linux
+      connectedhomeip/third_party/jsoncpp/repo/include
+      connectedhomeip/third_party/nlassert/repo/include
+      connectedhomeip/third_party/nlio/repo/include
+      connectedhomeip/third_party/inipp/repo/inipp
+      connectedhomeip/examples/lighting-app/lighting-common/include
+      ${NUTTX_APPS_DIR}/crypto/mbedtls/mbedtls/include)
+
+  set(MATTER_FLAGS
+      -DCHIP_HAVE_CONFIG_H
+      -fexceptions
+      -fpermissive
+      -std=c++17
```

Comment:
remove -fexceptions, -fpermissive, -std=c++17, this should be set from nuttx/libs/libxx/